### PR TITLE
Implement preview on highlighting quickpick in quick search

### LINKED ISSUE #191259: Implement preview on highlighting quickpick in quick search
It would be nice to show a preview editor that changes in the background based on which quick search item you have open. Furthermore, it'd be nice to highlight the result like we do in the symbol editor.

Image

This might be sort of non-trivial to tackle, as existing pickers that change the background editor only deal with a single open editor. In our case, we would need to implement a preview editor for this.

### PR DESCRIPTION
**Pull request by andreamah**
Fixes #191259

This implements a rough version of the preview on active picker selection.

https://github.com/microsoft/vscode/assets/31675041/79d4da71-ac14-43ba-93e2-a40b2d8fce2d

This has:
- Previewing on active selection
- Restoring state on escape

TODO:
- Try to get a 'true preview' editor state where we open a preview editor regardless of user settings.
- Omit editor navigation from back/forward history.

cc: @TylerLeonhardt @bpasero

### CODE REVIEW

File: src/vs/workbench/contrib/search/browser/anythingQuickAccess.ts
Comment (TylerLeonhardt) [inline src/vs/workbench/contrib/search/browser/anythingQuickAccess.ts:74]: I don't think this should live here. I don't think TextSearchQA should depend on anything in AnythingQA because AnythingQA is more of a gluing of other QA providers together into one "everything" experience

Comment (andreamah) [reply src/vs/workbench/contrib/search/browser/anythingQuickAccess.ts:74]: Where do you think this should live?

Comment (bpasero) [reply src/vs/workbench/contrib/search/browser/anythingQuickAccess.ts:74]: I agree, maybe in `src/vs/platform/quickinput/browser/quickAccess.ts`? Or even `src/vs/workbench/browser/quickaccess.ts` if only used in workbench.

**Comment by bpasero** [comment]
Saw this stack when testing:

```
ERR Cannot read properties of undefined (reading 'match'): TypeError: Cannot read properties of undefined (reading 'match')
    at UniqueContainer.value (vscode-file://vscode-app/Users/bpasero/Development/Microsoft/vscode/out/vs/workbench/contrib/search/browser/quickTextSearch/textSearchQuickAccess.js:69:26)
    at Emitter._deliver (vscode-file://vscode-app/Users/bpasero/Development/Microsoft/vscode/out/vs/base/common/event.js:889:26)
    at Emitter.fire (vscode-file://vscode-app/Users/bpasero/Development/Microsoft/vscode/out/vs/base/common/event.js:918:22)
    at UniqueContainer.value (vscode-file://vscode-app/Users/bpasero/Development/Microsoft/vscode/out/vs/platform/quickinput/browser/quickInput.js:650:51)
    at Emitter._deliver (vscode-file://vscode-app/Users/bpasero/Development/Microsoft/vscode/out/vs/base/common/event.js:889:26)
    at Emitter._deliverQueue (vscode-file://vscode-app/Users/bpasero/Development/Microsoft/vscode/out/vs/base/common/event.js:900:22)
    at Emitter.fire (vscode-file://vscode-app/Users/bpasero/Development/Microsoft/vscode/out/vs/base/common/event.js:923:22)
    at UniqueContainer.value (vscode-file://vscode-app/Users/bpasero/Development/Microsoft/vscode/out/vs/base/common/event.js:98:93)
    at Emitter._deliver (vscode-file://vscode-app/Users/bpasero/Development/Microsoft/vscode/out/vs/base/common/event.js:889:26)
    at Emitter._deliverQueue (vscode-file://vscode-app/Users/bpasero/Development/Microsoft/vscode/out/vs/base/common/event.js:900:22)
    at Emitter.fire (vscode-file://vscode-app/Users/bpasero/Development/Microsoft/vscode/out/vs/base/common/event.js:923:22)
    at vscode-file://vscode-app/Users/bpasero/Development/Microsoft/vscode/out/vs/base/common/event.js:98:93
    at vscode-file://vscode-app/Users/bpasero/Development/Microsoft/vscode/out/vs/base/common/event.js:1222:52
    at vscode-file://vscode-app/Users/bpasero/Development/Microsoft/vscode/out/vs/base/common/event.js:1235:37
    at Array.forEach (<anonymous>)
    at EventBufferer.bufferEvents (vscode-file://vscode-app/Users/bpasero/Development/Microsoft/vscode/out/vs/base/common/event.js:1235:20)
    at WorkbenchList.splice (vscode-file://vscode-app/Users/bpasero/Development/Microsoft/vscode/out/vs/base/browser/ui/list/listWidget.js:1187:32)
    at QuickInputList.setElements (vscode-file://vscode-app/Users/bpasero/Development/Microsoft/vscode/out/vs/platform/quickinput/browser/quickInputList.js:511:23)
    at QuickPick.update (vscode-file://vscode-app/Users/bpasero/Development/Microsoft/vscode/out/vs/platform/quickinput/browser/quickInput.js:790:30)
    at set items [as items] (vscode-file://vscode-app/Users/bpasero/Development/Microsoft/vscode/out/vs/platform/quickinput/browser/quickInput.js:402:18)
    at applyPicks (vscode-file://vscode-app/Users/bpasero/Development/Microsoft/vscode/out/vs/platform/quickinput/browser/pickerQuickAccess.js:89:34)
    at vscode-file://vscode-app/Users/bpasero/Development/Microsoft/vscode/out/vs/platform/quickinput/browser/pickerQuickAccess.js:112:52
    at applyFastAndSlowPicks (vscode-file://vscode-app/Users/bpasero/Development/Microsoft/vscode/out/vs/platform/quickinput/browser/pickerQuickAccess.js:114:27)
    at updatePickerItems (vscode-file://vscode-app/Users/bpasero/Development/Microsoft/vscode/out/vs/platform/quickinput/browser/pickerQuickAccess.js:178:27)
    at UniqueContainer.value (vscode-file://vscode-app/Users/bpasero/Development/Microsoft/vscode/out/vs/platform/quickinput/browser/pickerQuickAccess.js:206:59)
    at Emitter._deliver (vscode-file://vscode-app/Users/bpasero/Development/Microsoft/vscode/out/vs/base/common/event.js:889:26)
    at Emitter._deliverQueue (vscode-file://vscode-app/Users/bpasero/Development/Microsoft/vscode/out/vs/base/common/event.js:900:22)
    at Emitter.fire (vscode-file://vscode-app/Users/bpasero/Development/Microsoft/vscode/out/vs/base/common/event.js:923:22)
    at QuickPick.doSetValue (vscode-file://vscode-app/Users/bpasero/Development/Microsoft/vscode/out/vs/platform/quickinput/browser/quickInput.js:373:46)
    at UniqueContainer.value (vscode-file://vscode-app/Users/bpasero/Development/Microsoft/vscode/out/vs/platform/quickinput/browser/quickInput.js:558:26)
    at Emitter._deliver (vscode-file://vscode-app/Users/bpasero/Development/Microsoft/vscode/out/vs/base/common/event.js:889:26)
    at Emitter._deliverQueue (vscode-file://vscode-app/Users/bpasero/Development/Microsoft/vscode/out/vs/base/common/event.js:900:22)
    at Emitter.fire (vscode-file://vscode-app/Users/bpasero/Development/Microsoft/vscode/out/vs/base/common/event.js:923:22)
    at HistoryInputBox.onValueChange (vscode-file://vscode-app/Users/bpasero/Development/Microsoft/vscode/out/vs/base/browser/ui/inputbox/inputBox.js:354:31)
    at HTMLInputElement.<anonymous> (vscode-file://vscode-app/Users/bpasero/Development/Microsoft/vscode/out/vs/base/browser/ui/inputbox/inputBox.js:93:49)
```

**Comment by andreamah** [comment]
Fixed! ^

**Review by bpasero — APPROVED** [review]
I am giving +1 on a high level and defer to @TylerLeonhardt  for a deeper review. I suggest to tackle follow up things such as preview mode and history blocking on demand based on further feedback.

**Comment by andreamah** [comment]
I can probably add options when opening the editor to omit history and force preview, then figure out how to implement them on the editor side. Does that sound like the correct way to go?

**Comment by bpasero** [comment]
Yeah, given preview is on by default I think we can probably ship this for stable and then gather feedback based on users and then think about the implementation side of things in Feb?

**Comment by bpasero** [comment]
To be fair, I am not even sure if people might expect history to work or not. If you slowly jump over search results, maybe its expected that history allows you to navigate back? Thats why I am suggesting to wait for real user feedback!

**Comment by andreamah** [comment]
Talked to Tyler offline and he's ok with me merging for now and he'll review it after

**Review by Ros-Cos — COMMENTED** [review]
- [ ]  Smd
-  @

**Pull request merged**
